supervision/detection/core.py

```diff
@@ -1001,7 +1002,8 @@ def with_nms(
         self, threshold: float = 0.5, class_agnostic: bool = False
     ) -> Detections:
         """
-        Perform non-maximum suppression on the current set of object detections.
+        Perform non-maximum suppression on the current set
```

Comment:
Thanks for noticing the lack of description. I would replace it with something more precise. 

Performs non-max suppression on detection set. If the detections result from a segmentation model, the IoU mask is applied. Otherwise, box IoU is used.